src/main/java/seedu/address/logic/parser/AddLessonCommandParser.java

```diff
@@ -0,0 +1,40 @@
+package seedu.address.logic.parser;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+
+import seedu.address.logic.commands.AddLessonCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.lessons.Lesson;
+import seedu.address.model.person.Subject;
+
+
+
+/**
+ * Parses input arguments and creates a new AddLessonCommand object
+ */
+public class AddLessonCommandParser implements Parser<AddLessonCommand> {
+
+    // let be consistent with this, a model's optional field could be null, but during the rendering we check for that
+    @Override
+    public AddLessonCommand parse(String args) throws ParseException {
+        LocalTime startTime = TypeParsingUtil.parseTime("start", args);
+        LocalTime endTime = TypeParsingUtil.parseTime("end", args);
+        if (startTime.isAfter(endTime)) {
+            throw new ParseException("Start time must be before end time");
+        }
+        Subject subject = TypeParsingUtil.parseSubject("subject", args, true);
+        String studentName = TypeParsingUtil.parseStr("name", args);
+        LocalDate date = TypeParsingUtil.parseDate("day", args, true);
+        if (date == null) {
+            date = LocalDate.now();
+        }
+        LocalDateTime start = LocalDateTime.of(date, startTime);
+        LocalDateTime end = LocalDateTime.of(date, endTime);
+        if (subject == null) {
```

Comment:
Is there any reason why is it not compulsory for user to specify subject for adding lesson?

Reply:
Emm, because I feel that is a bit restrictive? Since the core function is schedule management, so maybe is good to only make time necessary?

Especially given our current implementation of subjects, which is enum type. When all of the five current subjects are not good suit, do not make sense to force the user to pick on in this case. For example, for those tuitioning h1 history, Olevel social study, or even cs2101

Reply:
Valid point, thanks for the clarification!